File: backend/app/schemas/agenda.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime


class AgendaEntryCreate(BaseModel):
    patient_id: UUID = Field(..., description="ID del paciente")
    facility_id: UUID = Field(..., description="ID de la facility (se valida que coincida con activeFacilityId)")
    seen_at: Optional[datetime] = Field(None, description="Fecha/hora de la atención (default: ahora)")
    note: Optional[str] = Field(None, description="Nota opcional sobre la atención")


class AgendaEntryUpdate(BaseModel):
    note: Optional[str] = Field(None, description="Nota opcional sobre la atención")
    seen_at: Optional[datetime] = Field(None, description="Fecha/hora de la atención")


class AgendaEntryResponse(BaseModel):
    id: UUID
    facility_id: UUID
    doctor_user_id: UUID
    patient_id: UUID
    seen_at: datetime
    note: Optional[str]
    created_at: datetime
    patient_name: Optional[str] = None  # Enriquecido con nombre del paciente
    doctor_name: Optional[str] = None  # Enriquecido con nombre del doctor
    
    class Config:
        from_attributes = True

File: frontend/src/types/agenda.ts
export interface AgendaEntry {
  id: string;
  facility_id: string;
  doctor_user_id: string;
  patient_id: string;
  seen_at: string; // ISO datetime string
  note: string | null;
  created_at: string; // ISO datetime string
  patient_name?: string; // Enriquecido desde backend
  doctor_name?: string; // Enriquecido desde backend
}

export interface AgendaEntryCreate {
  patient_id: string;
  facility_id: string;
  seen_at?: string; // ISO datetime string, opcional (default: ahora)
  note?: string | null;
}

export interface AgendaEntryUpdate {
  note?: string | null;
  seen_at?: string; // ISO datetime string, opcional
}